src/builder.rs

```diff
@@ -319,11 +351,7 @@ impl SaplingBuilder {
     /// This may be larger than the number of outputs that have been added to the builder,
     /// depending on whether padding is going to be applied.
     pub fn bundle_output_count(&self) -> usize {
-        // This matches the padding behaviour in `Self::build`.
-        match self.spends.len() {
-            0 => self.outputs.len(),
-            _ => std::cmp::max(MIN_SHIELDED_OUTPUTS, self.outputs.len()),
-        }
+        self.padding_rule.num_outputs(self.outputs.len())
```

Comment:
This is not a no-op for `PaddingRule::DEFAULT` because it stops taking into account `self.spends.len()`. This is a network-observable change, so we should do this in a (much smaller) crate release that we can deploy ideally as uniformly as possible across the ecosystem.